packages/mermaid/src/diagrams/git/gitGraphRenderer.js

```diff
@@ -338,26 +338,34 @@ const drawCommits = (svg, commits, modifyGraph) => {
 };
 
 /**
- * Detect if there are other commits between commit1's x-position and commit2's x-position on the
- * same branch as commit2.
+ * Detect if there are commits
+ * between commitA's x-position
+ * and commitB's x-position on the
+ * same branch as commitA, where
+ * commitA isn't main
  *
- * @param {any} commit1
- * @param {any} commit2
+ * @param {any} commitA
+ * @param {any} commitB
+ * @param branchToGetCurve
+ * @param p1
+ * @param p2
  * @param allCommits
- * @returns {boolean} If there are commits between commit1's x-position and commit2's x-position
+ * @returns {boolean}
+ * If there are commits between
+ * commitA's x-position
+ * and commitB's x-position
+ * on the source branch, where
+ * source branch is not main
+ * return true
  */
-const hasOverlappingCommits = (commit1, commit2, allCommits) => {
-  // Find commits on the same branch as commit2
-  const keys = Object.keys(allCommits);
-  const overlappingComits = keys.filter((key) => {
-    return (
-      allCommits[key].branch === commit2.branch &&
-      allCommits[key].seq > commit1.seq &&
-      allCommits[key].seq < commit2.seq
-    );
+const shouldRerouteArrow = (commitA, commitB, p1, p2, allCommits) => {
+  const commitBIsFurthest = dir === 'TB' ? p1.x < p2.x : p1.y < p2.y;
```

Comment:
It would be more portable if a function would not have side effects and closures like `dir`, but since it is only for this files and other anonymous functions do the same, that is ok.

Reply:
@nirname The alternative would be to have `commitBIsFurthest` (which relies on `dir`) calculated in `drawArrow` first, and then passed to `shouldRerouteArrow` instead of `p1, p2`. What do you think?

Reply:
I see. I don't know which approach is the right one, I would simply do:
```
const shouldRerouteArrow = (commitA, commitB, allCommits) => {
  const p1 = commitPos[commitA.id];
  const p2 = commitPos[commitB.id];
  const commitBIsFurthest = dir === 'TB' ? p1.x < p2.x : p1.y < p2.y;
  ...
```
simply because of arity reduction.